packages/medusa/src/loaders/config.ts

```diff
@@ -1,9 +1,21 @@
+import { getConfigFile } from "medusa-core-utils"
+import resolveCwd from "resolve-cwd"
 import { ConfigModule } from "../types/global"
-import { getConfigFile } from "medusa-core-utils/dist"
 import logger from "./logger"
 
 const isProduction = ["production", "prod"].includes(process.env.NODE_ENV || "")
 
+type ModuleDefinition = {
+  registration: string
```

Comment:
Comment: a module is supposed to represent a domain that we allow to replace entirely. It means that the module will in the future be able to hold more than a service and therefore I am not sure the registration name should be used to register a specific component in the container. We might need some other configuration to describe what a module contains and how they are registered in the container. A kind of descriptor like

`{ services: [{ registrationKey: formatRegistrationName(…), lifetime: SINGLETON }]  }`

in that case the modules configuration in medusa config should take that in count.

Let me know if I am going to far 😅

Reply:
I think this is a valid point. Though, in fear of doing premature optimizations, I'd like to stick with what we currently have and revisit in case this becomes relevant.

What do you think of that approach?

Reply:
I am alright with that but I just wanted to mention it as a reference for a later time. Also, in the actual way of doing it, a plugin allows to do the same so we will have to explain the future of it to the consumer to not have question such as « why having a module to do the same thing we can do with a plugin » and this is why the order of loading them is important.

at the moment creating a plugin that export a service named event-bus.ts already allow to override the internal event bus.